### PR TITLE
Retry temporary errors by default

### PR DESCRIPTION
**Pull request by rdelval**
* Reverting behavior of retry to retry every error by default.
* Added an option for users to enable failing in the case of temporary errors. This is useful for clients such as australis because they can give feedback to the user more quickly.
* Enabled Aurora URL validator along with tests. The validator will do its best to validate a URL and to correct it with some assumptions where possible. (Back ported from V2)
* Some cosmetic fixes for code style.

### CODE REVIEW

File: realis.go [hunk 1]
```diff
@@ -195,9 +199,9 @@ func BackOff(b Backoff) ClientOption {
 	}
 }
 
-func InsecureSkipVerify(InsecureSkipVerify bool) ClientOption {
+func insecureSkipVerify(insecureSkipVerify bool) ClientOption {
```
Comment (misho-kr) [inline realis.go]: Are you purposefully removing the access to this option?

File: util.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ func init() {
 	}
 }
 
-func validateAndPopulateAuroraURL(urlStr string) (string, error) {
+func validateAuroraURL(urlStr string) (string, error) {
```
Comment (misho-kr) [inline util.go]: The argument name does not have to indicate its type, it is right there next to it

File: realis.go (hunk 1 above)
Comment (rdelval) [reply realis.go]: I was not, thanks for catching this. Fixing it.

File: util.go (hunk 2 above)
Comment (rdelval) [reply util.go]: That's right, thanks for catching this.

**Comment by misho-kr** [comment]
The last commit has removed ".golangci.yml", is that correct?

**Comment by rdelval** [comment]
> The last commit has removed ".golangci.yml", is that correct?

It was but it's being returned in the next

**Pull request merged**